harness/determined/cli/model.py

```diff
@@ -79,15 +79,24 @@ def list_models(args: Namespace) -> None:
         render.tabulate_or_csv(headers, values, False)
 
 
+def model_by_name(args: Namespace) -> Model:
```

Comment:
should this logic be on the master side?

Reply:
@mapmeld pointed out the filter is master side, so this is fine with me. `len(models) > 1` is basically enforced by the db constraint, 2nd check doesn't hurt, len(models) == 0 just means nothing matched the filter, makes sense to be client side.

Reply:
@stoksc What do you mean the filter is master side?  I see logic that should be done in SQL being done in python code:

* get model id from name: one round trip with sql like `select id from models where name=$name`
* describe model using model id: another round trip with sql like `select * from models where id=$1`

Wheras you could do it with a single round-trip and a single query like `select * from models where name=$name`.

Reply:
@rb-determined-ai Is your objection more that (A) this requests the same info from the API twice, or (B) there should be an API endpoint to access model versions by name?

We are calling this only from `describe`, `list_versions`, and `register_version`
- In `describe` we make another call because `get_models` returns `num_versions` on each model but not specifics.
- In the other two, we need to call an API path which includes the model ID

Reply:
> we need to call an API path which includes the model ID

right, but we own the REST APIs.  If they are not suitable for the needs of our CLI they might not be suitable for end-users, either.

> Is your objection more that (A) this requests the same info from the API twice, or (B) there should be an API endpoint to access model versions by name?

I think (B).  I think it's an anitpattern to do some sql query, then feed output back into another sql query.  This is an antipattern even if you have direct access to the sql database (as the determined-master has) and it is a worse antipattern when you are doing it client-side.

I consider this a non-blocking issue. If you and @stoksc both understand my concern but don't share the concern, I'm fine landing this as-is.